### DISCUSSION TITLE
Event open/close inventory

### DISCUSSION THREAD

**Discussion by PARTOVIY** [SmartlyDressedGames/Unturned-3.x-Community#4263] [Ideas]
Nelson, hi.
When will the event for opening and closing inventory be implemented?
This is quite an important event for coders.

**Comment by cry-1337** [comment]
check class PlayerInteract and patch it, no problems 😎

**Reply by PARTOVIY** [reply]
PlayerInteract does not track inventory openings and closings.

**Reply by cry-1337** [reply]
any other class man

**Comment by Corbyncc** [comment]
The proper way to listen for inventory open/close is using PlayerAnimator.onGestureUpdated which has EPlayerGesture.INVENTORY_START/STOP respectively. AFAIK there are no other events or methods that get networked by the client besides this.

Its not super intuitive but its what we have, currently.

**Reply by QERT2002** [reply]
It's not a proper way.
Equip some gun in your hands and open or close inventory (PlayerAnimator.onGestureUpdated not see any changes in this situation)

**Reply by 0x59R11** [reply]
About opening the inventory I agree, but about closing it, it should always work.
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/41921371/38ddbe1f-109c-4543-8736-1b85d0669aa2)

**Comment by cry-1337** [comment]
you need to get inventory on "G"?

**Comment by GazziFX** [comment]
I think it can be relevant for your question https://github.com/SmartlyDressedGames/Unturned-3.x-Community/issues/2981#issuecomment-1005914125

**Reply by PARTOVIY** [reply]
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/49589187/83cf563a-884d-42e2-9314-32b89e7ab162)

### **The answer killed**

**Comment by sunnamed434** [comment]
Yep, QERT2002 is correct, this won't work once a client has equipped a weapon

My attempts:
- PlayerAnimator.onGestureUpdated (does not called at all)
- PlayerAnimator.onInventoryGesture (does not called when equipped a weapon)
- PlayerAnimator.OnGestureChanged_Global (does not called when equipped a weapon)

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/65300126/8f949431-4d55-4f9a-85f8-ea44525de4b6

**Comment by sunnamed434** [comment]
Solution: Patch ReceiveGestureRequest, this way you can know when inventory is opened/closed

**Reply by DanielWillett** [reply]
~~iirc that doesn't work if the player is holding a weapon.~~ nvm
